integrations/wpseo/wpseo.php

```diff
@@ -401,29 +396,36 @@ public function frontend_presentation( $presentation ) {
 	}
 
 	/**
-	 * Fixes the breadcrumb links and strings stored in the indexable table since Yoast SEO 14.0
+	 * Fixes the breadcrumb links and strings stored in the indexable table since Yoast SEO 14.0.
+	 *
+	 * In version 17.0, the breadcrumb links do not honor the filter `wpseo_dynamic_permalinks_enabled`.
 	 *
 	 * @since 2.8.3
 	 *
 	 * @param array $indexables An array of Indexable objects.
-	 * @return object
+	 * @return array
 	 */
 	public function breadcrumb_indexables( $indexables ) {
 		foreach ( $indexables as &$indexable ) {
-			switch ( $indexable->object_type ) {
-				case 'home-page':
-					$indexable->permalink = pll_home_url();
-					$indexable->breadcrumb_title = pll__( WPSEO_Options::get( 'breadcrumbs-home' ) );
-					break;
-
-				case 'post-type-archive':
-					if ( pll_is_translated_post_type( $indexable->object_sub_type ) ) {
-						$indexable->permalink = get_post_type_archive_link( $indexable->object_sub_type );
-						$breadcrumb_title = WPSEO_Options::get( 'bctitle-ptarchive-' . $indexable->object_sub_type );
-						$breadcrumb_title = $breadcrumb_title ? $breadcrumb_title : $indexable->breadcrumb_title; // The option may be empty.
-						$indexable->breadcrumb_title = pll__( $breadcrumb_title );
-					}
-					break;
+			// Handles both when the front page displays the list of posts or a static page.
+			if ( 'home-page' === $indexable->object_type || ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_on_front' ) === $indexable->object_id ) ) {
+				$indexable->permalink = pll_home_url();
+				$indexable->breadcrumb_title = pll__( WPSEO_Options::get( 'breadcrumbs-home' ) );
+			}
+
+			if ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_for_posts' ) === $indexable->object_id ) {
+				$indexable->permalink = get_permalink( $indexable->object_id );
+			}
+
+			if ( 'post-type-archive' === $indexable->object_type && pll_is_translated_post_type( $indexable->object_sub_type ) ) {
+				$indexable->permalink = get_post_type_archive_link( $indexable->object_sub_type );
+				$breadcrumb_title = WPSEO_Options::get( 'bctitle-ptarchive-' . $indexable->object_sub_type );
+				$breadcrumb_title = $breadcrumb_title ? $breadcrumb_title : $indexable->breadcrumb_title; // The option may be empty.
+				$indexable->breadcrumb_title = pll__( $breadcrumb_title );
+			}
```

Comment:
I'd suggest to use `break` at the end of each `if` statement, like this:
```suggestion
			if ( 'home-page' === $indexable->object_type || ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_on_front' ) === $indexable->object_id ) ) {
				$indexable->permalink = pll_home_url();
				$indexable->breadcrumb_title = pll__( WPSEO_Options::get( 'breadcrumbs-home' ) );
				break;
			}

			if ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_for_posts' ) === $indexable->object_id ) {
				$indexable->permalink = get_permalink( $indexable->object_id );
				break;
			}

			if ( 'post-type-archive' === $indexable->object_type && pll_is_translated_post_type( $indexable->object_sub_type ) ) {
				$indexable->permalink = get_post_type_archive_link( $indexable->object_sub_type );
				$breadcrumb_title = WPSEO_Options::get( 'bctitle-ptarchive-' . $indexable->object_sub_type );
				$breadcrumb_title = $breadcrumb_title ? $breadcrumb_title : $indexable->breadcrumb_title; // The option may be empty.
				$indexable->breadcrumb_title = pll__( $breadcrumb_title );
				break;
			}
```

Reply:
I don't believe that we should break the loop, as we may have, for example, a link to the homepage and a link to a CPT archive in the same breadcrumb. Adding `continue` instead may be a slight optimzation to avoid further tests.

Reply:
You're right, that's what I wanted to do. Monday morning after holidays... 😑

Reply:
I am proposing two optimizations.
1. The first one uses `elseif` instead of `continue` initially proposed, allowing to decrease teh number of tests in some cases.
```php
	public function breadcrumb_indexables( $indexables ) {
		foreach ( $indexables as &$indexable ) {
			// Handles both when the front page displays the list of posts or a static page.
			if ( 'home-page' === $indexable->object_type || ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_on_front' ) === $indexable->object_id ) ) {
				$indexable->permalink = pll_home_url();
				$indexable->breadcrumb_title = pll__( WPSEO_Options::get( 'breadcrumbs-home' ) );
			}

			elseif ( 'post' === $indexable->object_type && 'page' === $indexable->object_sub_type && get_option( 'page_for_posts' ) === $indexable->object_id ) {
				$indexable->permalink = get_permalink( $indexable->object_id );
			}

			elseif ( 'post-type-archive' === $indexable->object_type && pll_is_translated_post_type( $indexable->object_sub_type ) ) {
				$indexable->permalink = get_post_type_archive_link( $indexable->object_sub_type );
				$breadcrumb_title = WPSEO_Options::get( 'bctitle-ptarchive-' . $indexable->object_sub_type );
				$breadcrumb_title = $breadcrumb_title ? $breadcrumb_title : $indexable->breadcrumb_title; // The option may be empty.
				$indexable->breadcrumb_title = pll__( $breadcrumb_title );
			}

			elseif ( 'term' === $indexable->object_type && pll_is_translated_taxonomy( $indexable->object_sub_type ) ) {
				$indexable->permalink = get_term_link( $indexable->object_id );
			}
		}

		return $indexables;
	}
```
2. The second one returns to a `switch` as in the current version and is maybe the solution with the smallest number of tests. This comes with some price for the readability.
```php
	public function breadcrumb_indexables( $indexables ) {
		foreach ( $indexables as &$indexable ) {
			switch ( $indexable->object_type ) {
				case 'post':
					if ( 'page' !== $indexable->object_sub_type ) {
						break;
					}
					if ( get_option( 'page_for_posts' ) === $indexable->object_id ) {
						$indexable->permalink = get_permalink( $indexable->object_id );
						break;
					}
					if ( get_option( 'page_on_front' ) !== $indexable->object_id ) {
						break;
					}
					// Intentionally no break here as we are handling a front page (displaying posts).
				case 'home-page':
					$indexable->permalink = pll_home_url();
					$indexable->breadcrumb_title = pll__( WPSEO_Options::get( 'breadcrumbs-home' ) );
					break;
				case 'post-type-archive':
					if ( pll_is_translated_post_type( $indexable->object_sub_type ) ) {
						$indexable->permalink = get_post_type_archive_link( $indexable->object_sub_type );
						$breadcrumb_title = WPSEO_Options::get( 'bctitle-ptarchive-' . $indexable->object_sub_type );
						$breadcrumb_title = $breadcrumb_title ? $breadcrumb_title : $indexable->breadcrumb_title; // The option may be empty.
						$indexable->breadcrumb_title = pll__( $breadcrumb_title );
					}
					break;
				case 'term':
					if ( pll_is_translated_taxonomy( $indexable->object_sub_type ) ) {
						$indexable->permalink = get_term_link( $indexable->object_id );
					}
					break;
			}
		}

		return $indexables;
	}
```
What do you think?

Reply:
Even if I like `switch/case` statements in general, I don't think it's the right tool to use here, it decreases readability for the `'post'` case indeed.
Personally I prefer the first choice with `if/elseif`.